### PR TITLE
Firmware flash as container

### PR DESCRIPTION
**Pull request by joonas-fi**
**Do not merge** before https://github.com/tiiuae/px4-firmware/pull/99 is merged (this PR is built on top of)

When this is merged, commits to master trigger building images like `ghcr.io/tiiuae/px4-firmware:sha-29bb8cf`.

This container image, when run, flashes PX4 firmware to device. The container image contains all the built firmware files, so the container supports flashing multiple types of hardware.

### CODE REVIEW

File: .github/workflows/tiiuae-pixhawk.yaml [hunk 1]
```diff
@@ -132,3 +132,28 @@ jobs:
...
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=sha
+
+      - name: Firmware flasher - Prepare for build
+        # If we'd run build in Tools/ directory, I don't know how to pass both bin/ and Tools/ as context
+        run: |
+          cp px4-firmware/Tools/px_uploader.Dockerfile Dockerfile
```
Comment (joonas-fi) [inline .github/workflows/tiiuae-pixhawk.yaml:149]: This is a bit of a gross trick, so let me know if you have better ideas

Comment (jnippula) [reply .github/workflows/tiiuae-pixhawk.yaml:149]: You can use "file:" key to use dockerfile with custom name

```
with:
          push: true
          context: .
          file: px4-firmware/Tools/px_uploader.Dockerfile
          tags: ${{ steps.containermeta.outputs.tags }}
          labels: ${{ steps.containermeta.outputs.labels }}
```

Comment (joonas-fi) [reply .github/workflows/tiiuae-pixhawk.yaml]: Nice, thanks!

I knew you could give the file, but assumed build context was relative to that, but apparently the `context: .` keeps the context at repo root, so everything works out!

Thanks again!

**Comment by jnippula** [comment]
There are some changes in master now, this needs yo be rebased. Otherwise ok for me

**Review by jnippula — APPROVED** [review]

**Pull request merged**